File: Java/0001-1000/001-100/21-30/27.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int swap, now = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                nums[now] = nums[i];
                now += 1;
            }
        }
        return now;
    }
}

File: Python/0001-1000/001-100/21-30/27.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        c = 0
        if nums:
            beg = 0
            end = len(nums) - 1
            while end >= 0 and nums[end] == val:
                c += 1
                end -= 1
            while beg <= end:
                if nums[beg] != val:
                    beg += 1
                else:
                    nums[beg], nums[end] = nums[end], nums[beg]
                    beg += 1
                    while end >= 0 and nums[end] == val:
                        c += 1
                        end -= 1
        return len(nums) - c
